### PR TITLE
sys/net/routing/nhdp: get rid of vtimer

### PR DESCRIPTION
**Pull request by kYc0o**
Splitted first part of PR #4660

### CODE REVIEW

File: sys/net/routing/nhdp/nhdp.c [hunk 1]
```diff
@@ -100,7 +101,10 @@ kernel_pid_t nhdp_start(void)
         /* Configure periodic timer message to refresh metric values */
         if (nhdp_pid != KERNEL_PID_UNDEF) {
             metric_interval = timex_from_uint64(DAT_REFRESH_INTERVAL * SEC_IN_USEC);
-            vtimer_set_msg(&metric_timer, metric_interval, nhdp_pid, NHDP_METRIC_TIMER, NULL);
+            metric_msg.type = NHDP_METRIC_TIMER;
+            metric_msg.content.ptr = NULL;
+            xtimer_set_msg64(&metric_timer, timex_uint64(metric_interval),
+                               metric_msg, nhdp_pid);
```
Comment (cgundogan) [inline sys/net/routing/nhdp/nhdp.c]: indentation

File: sys/net/routing/nhdp/nhdp.h [hunk 2]
```diff
@@ -90,7 +90,7 @@ extern "C" {
  */
 typedef struct nhdp_if_entry_t {
     kernel_pid_t if_pid;                        /**< PID of the interface's handling thread */
-    vtimer_t if_timer;                          /**< Vtimer used for the periodic signaling */
+    xtimer_t if_timer;                          /**< Xtimer used for the periodic signaling */
```
Comment (cgundogan) [inline sys/net/routing/nhdp/nhdp.h]: I would name it `xtimer` and not `Xtimer`

Comment (cgundogan) [inline sys/net/routing/nhdp/nhdp.h]: `Xtimer` reminds me of `X-Men`..

Comment (kYc0o) [inline sys/net/routing/nhdp/nhdp.h]: And you don't like `X-Men`?

**Comment by cgundogan** [comment]
looks great! please squash

**Comment by kYc0o** [comment]
squashed

**Comment by cgundogan** [comment]
ACK

File: sys/net/routing/nhdp/nhdp.c [hunk 3]
```diff
@@ -100,7 +101,10 @@ kernel_pid_t nhdp_start(void)
         /* Configure periodic timer message to refresh metric values */
         if (nhdp_pid != KERNEL_PID_UNDEF) {
             metric_interval = timex_from_uint64(DAT_REFRESH_INTERVAL * SEC_IN_USEC);
-            vtimer_set_msg(&metric_timer, metric_interval, nhdp_pid, NHDP_METRIC_TIMER, NULL);
+            metric_msg.type = NHDP_METRIC_TIMER;
```
Comment (cgundogan) [inline sys/net/routing/nhdp/nhdp.c]: is my font displaying something wrong here, or is this and the subsequent lines really once off by one white-space?

Comment (kYc0o) [inline sys/net/routing/nhdp/nhdp.c]: It's your screen, the lines are subsequent without any white space

**Comment by kYc0o** [comment]
Murdock fails

**Comment by kYc0o** [comment]
Murdock failed because a define from `vtimer.h` was missing. Added to `nhdp.c` since this is the only occurence on RIOT.

**Comment by kYc0o** [comment]
Now Murdock is happy :dancer:

**Comment by authmillenon** [comment]
then let's go.

**Comment by kYc0o** [comment]
Thanks Martine! :)

**Pull request merged**